data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -139,9 +213,18 @@ public static List<DataFile> listPartition(
                   buildDataFile(fileStatus.get(index), partitionKey, spec, metrics, "orc");
             });
       } else {
-        throw new UnsupportedOperationException("Unknown partition format: " + format);
+        if (skipOnError) {
+          LOG.warn("Skipping unknown partition format: {} - {}", format, partitionUri);
+        } else {
+          throw new UnsupportedOperationException(
```

Comment:
it looks like this part is actually never hit in the tests and I wonder how difficult it would be to have a test that would hit this exception

Reply:
UT `testMigrateUnknownFormatTableSkipOnError` was added to try to cover this case.

Reply:
thanks